packages/calcite-components/src/components/input-number/input-number.tsx

```diff
@@ -1059,8 +1085,8 @@ export class InputNumber
         autofocus={this.el.autofocus ? true : null}
         defaultValue={this.defaultValue}
         disabled={this.disabled ? true : null}
-        enterKeyHint={this.el.enterKeyHint}
-        inputMode={this.el.inputMode}
+        enterKeyHint={this.el.enterKeyHint || this.el.getAttribute("enterkeyhint")}
```

Comment:
This looks a little odd to me because you could just do

```tsx
enterKeyHint={this.enterKeyHint}
```

But I get why its like the way you have. So its easier to remove the property in the future breaking change release 👍

Reply:
This is actually needed since the component prop won't be populated if I set the global attribute. The custom props override the native behavior, so the prop will not be set when using the native global attributes.